sdk/storage/azure-storage-blob/azure/storage/blob/_upload_helpers.py

```diff
@@ -316,12 +326,12 @@ def upload_append_blob(  # pylint: disable=unused-argument
                 append_position_access_conditions=append_conditions,
                 progress_hook=progress_hook,
                 headers=headers,
-                **kwargs)
+                **kwargs))
         except HttpResponseError as error:
-            if error.response.status_code != 404:
+            if error.response.status_code != 404:  # type: ignore
                 raise
             # rewind the request body if it is a stream
-            if hasattr(stream, 'read'):
+            if stream is not None and hasattr(stream, 'read'):
```

Comment:
Just making a note here that this is why I added this check:
`azure\storage\blob\_upload_helpers.py:337: error: Item "None" of "Optional[IO[Any]]" has no attribute "seek"  [union-attr]`

Reply:
Make the stream non-optional and then get rid of the none-check